### PR TITLE
community: Add MarkdownifyTransformer to langchain_community.document_transformers

### PR DESCRIPTION
**Pull request by sokolgood**
- Added new document_transformer: MarkdonifyTransformer, that uses `markdonify` package with customizable options to convert HTML to Markdown. It's similar to Html2TextTransformer, but has more flexible options and also I've noticed that sometimes MarkdownifyTransformer performs better than html2text one, so that's why I use markdownify on my project. 
- Added docs and tests

- Usage:
```python
from langchain_community.document_transformers import MarkdownifyTransformer

markdownify = MarkdownifyTransformer()
docs_transform = markdownify.transform_documents(docs)
```

- Example of better performance on simple task, that I've noticed:
```
<html>
<head><title>Reports on product movement</title></head>
<body>
<p data-block-key="2wst7">The reports on product movement will be useful for forming supplier orders and controlling outcomes.</p>
</body>
```
**Html2TextTransformer**: 
```python
[Document(page_content='The reports on product movement will be useful for forming supplier orders and\ncontrolling outcomes.\n\n')]
# Here we can see 'and\ncontrolling', which has extra '\n' in it
```
**MarkdownifyTranformer**:
```python
[Document(page_content='Reports on product movement\n\nThe reports on product movement will be useful for forming supplier orders and controlling outcomes.')]
```

### CODE REVIEW

**Comment by mayb333** [comment]
@efriis

**Comment by mayb333** [comment]
@baskaryan

**Comment by mayb333** [comment]
@eyurtsev @baskaryan

**Comment by mayb333** [comment]
@efriis @hwchase17

File: .gitignore [hunk 1]
```diff
@@ -147,7 +147,7 @@ wandb/
 
 # asdf tool versions
 .tool-versions
-/.ruff_cache/
+.ruff_cache/
```
Comment (hwchase17) [inline .gitignore:150]: this probably shouldnt be touched
```suggestion
/.ruff_cache/
```

**Review by hwchase17 — COMMENTED** [review]
a integration document would be nice!

**Comment by mayb333** [comment]
@hwchase17 Added integration document and changed a little bit unit test that was failing.

> !!! Need to specify `markdownify` == 0.12.1 for successfully computing unit tests for MarkdownifyTransformer

pyproject.toml has this:
```
[tool.poetry.dependencies]
markdownify = {version = "^0.11.6", optional = true}
```

I doubt that I update version correctly, so could you pls do it - **Update markdownify to 0.12.1**

**Review by hwchase17 — APPROVED** [review]

**Comment by mayb333** [comment]
@hwchase17 
Removed unit-test that was failing. 

It was failing because of the version of `markdownify`
```
[tool.poetry.dependencies]
markdownify = {version = "^0.11.6", optional = true}
```

It should be 0.12.1 for successful unit-test `test_remove_style`

**Pull request merged**